vignettes/delay_distributions.Rmd

```diff
@@ -0,0 +1,155 @@
+bibliography: resources/library.json
+link-citations: true
+vignette: >
+  %\VignetteIndexEntry{A primer on working with delay distributions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette is intended to be guidance for working with probability distributions in R in the context of using delay distributions with _cfr_ to obtain delay-corrected estimates of disease severity.
+
+```{r setup}
+# load necessary packages
+library(cfr)
+
+# some distribution packages
+library(distributional)
+library(distcrete)
+```
+
+## A brief primer on distributions in R
+
+R provides rich and extensive support for representing and working with probability distributions, and this can be seen from the [CRAN probability distributions task view](https://cran.r-project.org/view=Distributions).
+
+Users might already be familiar with some distributions and their related functionality --- such as the probability density function, or random number generation --- provided in the _stats_ package which is loaded when R is started.
+
+For example, the Gamma distribution's probability density function (PDF) is represented by `stats::dgamma()`.
+
+```{r}
+# the probability density function at `x` for a Gamma distribution
+dgamma(x = seq(10), shape = 5, rate = 1)
+```
+
+## Using delay distribution densities in _cfr_
+
+To correct for reporting delays in disease severity estimation, we are primarily interested in the PDF or PMF (probability mass function) of the distribution of reporting delays between cases and known outcomes.
+
+We refer to the _R functions providing_ both the PDF and PMF as the _density_ of the distribution.
+
+The delay distribution density must be passed to functions such as `cfr_static()` or `cfr_time_varying()` via the `delay_density` argument. This can be represented in pseudo-code as
+
+```
+cfr_*(data, delay_density = <DENSITY_FUNCTION>)
+```
+
+## Preparing delay distribution density for _cfr_
+
+_Importantly_, the _cfr_ functions must receive the delay density in such a way that the density can be calculated over a flexible number of values (the sequence of days in the outbreak data).
+
+For example, the density function for a Gamma distribution can be made generalisable to any set of values, by wrapping it within another function that accepts a vector of numbers.
+
+```{r}
+# wrap stats::dgamma() in a function
+# the Gamma distribution parameters are contained within dens_gamma()
+dens_gamma <- function(x) {
+  stats::dgamma(x = x, shape = 5, scale = 1)
+}
+
+# check over a vector of `x`
+dens_gamma(x = seq(10))
+```
+
+::: {.alert .alert-info}
+More information on working with functions, and especially anonymous functions, can be found in the [chapter on Functional Programming in _Advanced R_](https://adv-r.hadley.nz/fp.html).
+Users working with R > 4.1.0 can also use the new syntax for anonymous functions.
```

Comment:
Is it worth showing an example of the new syntax in parentheses?

Reply:
Sure, will add.

Reply:
Added in https://github.com/epiverse-trace/cfr/pull/88/commits/96bcad74cf4e9dc0b1d3e4a30d06071c18485505